### PR TITLE
Create Hamiltonian_path.java

### PR DESCRIPTION
**Pull request by iianuragii**
## 🛠️ Issue (Number)

Issue No. #339 

## 👨‍💻 Changes proposed

This is the code for Hamiltonian Path in java which was assigned to me.

## ✔️ Check List (Check all the applicable boxes) 

- [✔️ ] My code follows the code style of this project.
- [✔️ ] This PR does not contain plagiarized content.
- [✔️ ] The title of my pull request is a short description of the requested changes.

## 📄 Note to reviewers

Hamiltonian Path is a path in an undirected graph that visits every vertex exactly once.
SAMPLE OUTPUT:-
image

## 📷 Screenshots

image

### CODE REVIEW

File: Java/Graphs/Hamiltonian_path.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    private int[][] graph;
+ 
+    /** Function to find cycle **/
+    public void findHamiltonianCycle(int[][] g)
+    {
+        V = g.length;
+        path = new int[V];
+ 
+        Arrays.fill(path, -1);
+        graph = g;
```
Comment (siddhi-244) [inline Java/Graphs/Hamiltonian_path.java:18]: Try to add some comments here expressing what you are actually doing !

**Review by siddhi-244 — CHANGES_REQUESTED** [review]

**Comment by iianuragii** [comment]
> @iianuragii Nice code Add a screenshot of the code output

I have added it in the Issue section of "Note to Reviewers".

**Review by siddhi-244 — APPROVED** [review]

**Review by 072arushi — CHANGES_REQUESTED** [review]
Add comments to explain your algorithm and time complexity

File: Java/Graphs/Hamiltonian_path.java (hunk 1 above)
Comment (iianuragii) [reply Java/Graphs/Hamiltonian_path.java:18]: Done

**Comment by iianuragii** [comment]
> Add comments to explain your algorithm and time complexity

Comments are already added

**Review by Arun9739 — APPROVED** [review]
Approved !

**Review by Rishikavishnoi — APPROVED** [review]

**Review by 072arushi — APPROVED** [review]

**Comment by iianuragii** [comment]
Hiiii Sir,
My code is approved by four of the reviewers to whom you have given the responsibility to check. Can you please merge my request if there is not any further problems to overwrite.
Thank You.

**Review by Kumar-laxmi — APPROVED** [review]

**Pull request merged**